Lib/fontTools/pens/pointPen.py

```diff
@@ -60,6 +61,21 @@ def addComponent(
         """Add a sub glyph."""
         raise NotImplementedError
 
+    def addVarComponent(
+        self,
+        glyphName: str,
+        transformation: DecomposedTransform,
+        location: Dict[str, float],
+        identifier: Optional[str] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Add a VarComponent sub glyph. The 'transformation' argument
+        must be a DecomposedTransform from the fontTools.misc.transform module,
+        and the 'location' argument must be a dictionary mapping axis tags
+        to their locations.
+        """
+        raise NotImplementedError
```

Comment:
you made the AbstractPen (segment-oriented protocol) raise AttributeError, but here in the AbstractPointPen you are raising NotImplementedError.. you should do the same for both respective's addVarComponent methods

Reply:
since the TTGlyphSet now only catches AttributeError, and not also NotImplementedError, it means that a pointpen that inherits from AbstractPointPen but doesn't implement addVarComponent will now crash when used with the new varComponents-aware TTGlyphSet